### ISSUE TITLE
[Due for payment 2025-02-28] [$250] 1:1 Submit expense - The receiving end of the IOU should not be asked to fill out information if the scan failed

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#51574]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.54
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5121232&group_by=cases:section_id&group_order=asc&group_id=294998
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Start the Submit expense flow in a 1:1 conversation with a user you have access to
2. Select the scan option
3. Upload a picture that will fail the scan process
4. Wait for the scan to fail
5. Log in as the other account
6. Navigate to the IOU that failed the scan
7. Verify the receiver is not asked to fill out the missing information

## Expected Result:
The receiving end of the IOU should not be asked to fill out information if the scan failed

## Actual Result:
The receiving end of the IOU is asked to fill out information if the scan failed but can't

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/7317ab19-6b6a-4337-91c2-e925dd5af6f4

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021854642765194540226</li>
        <li>Upwork Job ID: 1854642765194540226</li>
        <li>Last Price Increase: 2024-11-14</li>
<li>Automatic offers: </li>
<ul>
<li>FitseTLT | Contributor | 105019830</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @alexpensify</details>

**Comment by lanitochka17** [comment]
@alexpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
1:1 Submit expense - Scan failure should not ask receiver to enter information

### What is the root cause of that problem?
We are displaying the same `Enter an amount` text for both side of the users
https://github.com/Expensify/App/blob/c3dde46202d2f48934ef7aa617d69c8b4c0b5ff6/src/components/ReportActionItem/MoneyRequestView.tsx#L272
### What changes do you think we should make in order to solve the problem?

We should display this copy only if the user `canEditAmount` and add another appropriate copy (like `No amount set yet`) and display it when the user can't modify the amount 
```
                    translationPath: canEditAmount ? 'common.error.enterAmount' : 'common.error.noAmount',

```
We should do the same `enterDate` and `enterMerchant`
### What alternative solutions did you explore? (Optional)
Optionally we can also avoid displaying the error for the receiving end

**Comment by alexpensify** [comment]
No update yet

**Comment by alexpensify** [comment]
@getusha - can you please confirm if this proposal will fix this issue? 

Heads up, I will be offline until Tuesday, November 12, 2024, and will not actively watch over this GitHub during that period.

If this GitHub requires an urgent update, please ask for help in the #expensify-open-source Slack Room. If it can wait, I'll continue the review process when I return online. Thanks!

**Comment by FitseTLT** [comment]
@alexpensify The title of the issue should be updated according to the expected result 

>Expected Result:
The receiving end of the IOU should not be asked to fill out information if the scan failed

**Comment by alexpensify** [comment]
Done! Thanks for the feedback.

**Comment by alexpensify** [comment]
@getusha any update regarding the review here? Thanks!

**Comment by alexpensify** [comment]
@getusha if you are unable to review. I'll start the process of reassigning on Monday.

**Comment by mountiny** [comment]
What is the correct behaviour here, I am not 100% sure

**Comment by mountiny** [comment]
Yeah I think its a bit weird to show the RBR if you cannot edit the expense. Curious for @trjExpensify throughts

**Comment by mountiny** [comment]
> That said, I think we need to bring back that system message in the expense thread, to communicate why this is $0.00.

do we have an issue for bringing that back?

**Comment by trjExpensify** [comment]
I believe it was going to come with Mills' sys message doc, is that right Jason?

**Comment by JmillsExpensify** [comment]
Yeah, that's right.

**Comment by mountiny** [comment]
Alright so for the purposes of this issue, we should make sure no error is shown on the side that cannot edit it.

@getusha can you please choose the proposal according to that expectation? thanks!

**Comment by FitseTLT** [comment]
@JmillsExpensify @mountiny @getusha Before I make changes I want a confirmation:
RBRs are supposed to be directing the user into fixing errors so in addition to removing error and violation messages in transaction thread details page for the non-editable side we also need to remove the RBR in LHN and report previews. 
So are u ok with it or want to only change the error text from `Enter an amount` to `Amount is not entered yet` for the non-editable side? Thx!

**Comment by FitseTLT** [comment]
> @FitseTLT we agreed on showing a system message instead of an error on the input, for this issue we just need to remove the error and RBR for the non-editable side.
> 
> @mountiny should we show `$0.00` instead of leaving the field empty?

With the RBR I think in addition to the RBR shown in transaction details view, we should also remove the one displayed in LHN and ReportPreview and also MoneyRequestPreview for the non-editable side. Otherwise, we will be leading the user with the RBR into transaction details view where there won't be any violation RBR displayed which would be confusing.
@getusha BTW there are similar problems regarding other violations for other fields as well (like `Missing Tag`) which would extend the scope of this issue so should we handle only the amount and merchant case here?

**Comment by mountiny** [comment]
Correct, for the user who cannot edit the expense, they should not see the RBR anywhere

> which would extend the scope of this issue so should we handle only the amount and merchant case here?

Yeah I assume this applies for any violation, the fix should cover a generic case of not showing the RBR when the user cannot edit the expense

**Comment by mountiny** [comment]
@getusha @FitseTLT How is this looking?

**Comment by FitseTLT** [comment]
PR will be ready tomorrow 👍

**Comment by FitseTLT** [comment]
@mountiny I have been working on removing RBR from the non-editable side but the problem is RBR is shown on a different level such as workspace chat, money report, report preview, and money request preview. So let's say the violation/error is related to the `amount` field to determine whether the current user can edit `amount` field of a transaction we need to have the parentReportAction and transactionThreadReport linked to the transaction
https://github.com/Expensify/App/blob/64eaf2fdd357bd06008adfb9c0c049e09cd375b9/src/components/ReportActionItem/MoneyRequestView.tsx#L153-L157
but the problem is we are not always guaranteed these data is available in onyx. For instance, if the user just logged in and didn't open the transaction Thread reports when we try to calculate the RBR for the workspace chat the parent report action linked to the transactions will not be available until they open the expense report so we can't reliably and consistently determine canEditMoneyRequest.
I think all we can do is show different copy in transaction thread based on whether or not the user can edit the field as in transaction thread details view we can determine whether the field can be edited. WDYT

**Comment by mountiny** [comment]
@FitseTLT I see what you mean about not all data being accessible. So what we can do is to change the text for the admin, but they will still see the RBR leading you to something they cannot edit if they just signed in, is that right?

**Comment by FitseTLT** [comment]
> @FitseTLT I see what you mean about not all data being accessible. So what we can do is to change the text for the admin, but they will still see the RBR leading you to something they cannot edit if they just signed in, is that right?

Yes

**Comment by FitseTLT** [comment]
@mountiny bump on the ^

**Comment by mountiny** [comment]
I think that is ok edge case to have for now, I believe we will stop sending the expense reports without RBR to the user from the backend

So lets just hide the text

**Comment by FitseTLT** [comment]
> I think that is ok edge case to have for now, I believe we will stop sending the expense reports without RBR to the user from the backend
> 
> So lets just hide the text

@mountiny To summarize:
RBRs on workspace chats, report previews, money request view will show for non-editable side (b/c we can't reliably determine whether it is editable or not due to lack of data) and when the user opens the expense details page we will only have RBR icons on the fields but not text beside it ??

**Comment by mountiny** [comment]
No, once they open the thread, the RBR will be gone, as we will have enough data.

Users are not expected to keep signing out so this is an edge case for only fresh sign ins, if you use the app you get all the data through pusher and RBR should not show up

**Comment by FitseTLT** [comment]
Working on this. Will make PR ready by EOW 👍

**Comment by alexpensify** [comment]
Ok, thank you for the update.

**Comment by alexpensify** [comment]
@FitseTLT do you have a PR update? Thanks!

**Comment by FitseTLT** [comment]
> @FitseTLT do you have a PR update? Thanks!


Done with the code will record screenshots and create the PR tomorrow. Thx for your patience

**Comment by alexpensify** [comment]
## Heads up, I will be offline until Wednesday, December 18, 2024, and will not actively watch over this GitHub during that period.

If this GitHub requires an urgent update, please ask for help in the #expensify-open-source Slack Room. If the inquiry can wait, I'll review it when I return online.

**Comment by alexpensify** [comment]
Weekly Update: PR is moving along through the review process

**Comment by alexpensify** [comment]
## Heads up, I will be offline until Friday, January 3, 2025, and will not actively watch over this GitHub during that period.

If this GitHub requires an urgent update, please ask for help in the #expensify-open-source Slack Room. If the inquiry can wait, I'll review it when I return online.

**Comment by alexpensify** [comment]
Weekly Update: PR is in review

**Comment by alexpensify** [comment]
Weekly Update: Lots of changes in the PR, so going through more review

**Comment by alexpensify** [comment]
@getusha and @mountiny looks like we are still reviewing the PR. Is there anything we can do to complete it or just need more time?

**Comment by mountiny** [comment]
@alexpensify I will come back to this tomorrow, we will need to take a different approach to ensure we dont hurt performance

**Comment by FitseTLT** [comment]
@mountiny Can you tag someone to get us the correct text equivalent of 
```
Enter a merchant name
Enter an amount
Enter a date
```
from the user side who cannot edit the expense?

**Comment by mountiny** [comment]
@LLPeckham Welcome back! Could you please help us with a copy of these errors? The situation now is that user can see the erros on the expense/request (when smartscan fails for example) and we ask them to:
- `Enter a merchant name`
- `Enter an amount`
- or `Enter a date`

This is fine for the person who can edit it, but for the person who looks at the expense and cannot edit it, we should change the copy to just inform them that these details have to be inserted by the requestor (this is mainly in a P2P case) before the expense can be paid.

Maybe something like:
- `Missing merchant name` and similar should work well

**Comment by FitseTLT** [comment]
@getusha Can you please review the PR?

**Comment by alexpensify** [comment]
Contributor: @FitseTLT paid $250 via Upwork
Contributor+: @getusha owed $250 via NewDot - please submit a request in Chat

**Comment by garrettmknight** [comment]
$250 approved for @getusha
